src/markov/markov_approx.jl

```diff
@@ -130,3 +130,50 @@ function _rouwenhorst(p::Real, q::Real, m::Real, Δ::Real, n::Integer)
         return linspace(m-Δ, m+Δ, n), θN
     end
 end
+
+
+"""
+Accepts the simulation of a discrete state Markov chain and estimates
+the transition probabilities via maximum likelihood.
+
+Note: Only states that are observed in the history will be included
+in the estimation.
+
+##### Arguments
+
+- `X::Vector{T}` : Simulated history of Markov states
+
+##### Returns
+
+- `mc::MarkovChain{T}` : A Markov chain holding the state values and
+transition matrix
+
+"""
+function estimate_MC_discrete{T}(X::Vector{T})
+    # Get length of simulation
+    capT = length(X)
+
+    # Find all unique observations and sort them.
+    states = sort!(unique(X))
+    nstates = length(states)
+
+    # Counter matrix
+    cm = zeros(nstates, nstates)
+
+    # Compute conditional probabilities for each state
+    state_i = findfirst(states, X[1])
+    for t in 1:capT-1
+        # Find next period's state
+        state_j = findfirst(states, X[t+1])
```

Comment:
I think you are ok with `searchsortedfirst` here because you _know_ any `X[t+1]` that can come up appears exactly once in `states` 

But, I take your point. With the small examples we have in the tests I would assume there is no performance difference.